usdot-fhwa-stol/carma-streets
# Fixes for tsc_service failed initilaization

# PR Details
## Description


+ Executable could not find librkafka library (fixed with ldconfig call)
+ Added find package to find kafka-clients library
+ Update main to not attempt to initialize logger since that now happens inside service base (This was triggering singleton exception)
+ Updated docker-compose with correct environment variables for V2X-Hub and tsc_service
+ Added javadoc comments to service base classes
+ Added try catch to initialize method to ensure no exceptions are uncaught
+ Update streets_service to launch time consumer as detached thread to avoid blocking base class start method or thread to go out of scope in inherited start method.

## Related Issue
#324






## Motivation and Context
XIL CARMA-Streets Simulation Integration

Simulation Integration with CARMA-Streets under XIL
## How Has This Been Tested?




Integration tested by deploying V2X-Hub and CARMA Streets.
1) Deploy V2X-Hub in non-simulation mode and forward MAP message to kafka using the CARMA-Streets Plugin
2) Deploy Intersection Model with an OSM map that shares an intersection id with the MAP message
3) Deploy EVC with no spat forwarding configurations (1 2 7 menu option all set to zeros)
4) Send snmpset -v1 -c administrator <ip-address>:SNMP-Port  1.3.6.1.4.1.1206.3.5.2.9.44.1.0 i 2 to enable spat ( do not use local host as IP)
5) Configure tsc_service to listen for spat at 6053 and send NTCIP commands to ip-address:SNMP-Port. ( do not use local host as IP)
6) Deploy TSC Service in simulation mode with the following environment variables set
      SIMULATION_MODE: TRUE
      CONFIG_FILE_PATH: ../manifest.json
      TIME_SYNC_TOPIC: time_sync
7) Use docker exec -it kafka bash to ssh into kafka container
8) Use kafka-console-producer --bootstrap-server=127.0.0.1:9092 --topic=time_sync to send sample time sync messages
{ "timestep":1231,"seq":1231322}
9) This should trigger tsc_service to produce SPaT JSON on kafka topic modified_spat

## Types of changes




## Checklist:
